src/main/java/sparksoniq/semantics/visitor/AbstractExpressionOrClauseVisitor.java

```diff
@@ -75,19 +75,19 @@
 
 public abstract class AbstractExpressionOrClauseVisitor<T> {
 
-    public T visit(ExpressionOrClause expression, T argument) {
+    public T visit(Node expression, T argument) {
```

Comment:
I have a feeling "Node expression" naming can be confusing down the line as we have an Expression class as well. How about naming the arguments as "Node node" in all similar places where we have refactored from ExpressionOrClause